### ISSUE TITLE
[feature request] Option to have both titlebars and top borders on stacked / tabbed layouts

### ISSUE THREAD

**Issue by avrelaun** [i3/i3#2738]
I use colored window borders as a way to make me more aware of the currently active window.

In tabbed & stacked layouts, the top borders are replaced with titlebars, making the borders of the window less obvious after you've open a lot of siblings.

It would be nice nice to have an option for having both the titlebars **_and_** the top borders, making the borders completely surround the currently active window.

This would also need to comply with the hide_edge_borders smart option: If the current workspace only has one main container with tabbed or stacked childs, the behaviour would be the same as of today (no borders sides, top and bottom).

Stacking layout as of today:
![stacked-before](https://cloud.githubusercontent.com/assets/412102/24966251/b2a51408-1fa6-11e7-8683-b72d20125dd3.png)
Stacking layout as proposed:
![stacked-after](https://cloud.githubusercontent.com/assets/412102/24966249/b2a2c5fe-1fa6-11e7-95b6-8f36a5461034.png)

Tabbed layout as of today:
![tabbed-before](https://cloud.githubusercontent.com/assets/412102/24966250/b2a32f1c-1fa6-11e7-9d1a-b15b51ef3419.png)
Tabbed layout as proposed:
![tabbed-after](https://cloud.githubusercontent.com/assets/412102/24966248/b2a23760-1fa6-11e7-9487-274a66ad0d2d.png)

**Comment by Airblader** [comment]
I understand why this appeals to you and I thought about this myself several times. However, this is just the look & feel of i3 and we don't want to make every detail about that configurable. Is there any real, practical benefit here that would make this something more than an eye-candy topic?

**Comment by avrelaun** [comment]
Eye-candy is a nice side-effect, but this is really about being more conscious of the currently active terminal (this is why I configured a flashy color and a 2px border). I often end up with 10+ terminals on several workspaces / screens, and with a few tabs the current window is less noticeable.

If you and I thought about it, chances are others i3 users also did and would benefit from it.
I don't think that would require much lines of code, and I can contribute.

**Comment by Airblader** [comment]
It's not about the lines of code, it's about the mental complexity of making every detail of the look & feel configurable. I still don't quite see what the actual benefit of four borders is over three borders. Why does it become so much more noticeable that way?

**Comment by avrelaun** [comment]
A window fully surrounded with borders is way more noticeable on a very busy workspace.
I do not know what I could do more to explain, maybe more screenshots ?

**Comment by Airblader** [comment]
The visual change is clear, but I currently don't think the effect is big enough to justify the cost of adding complexity for this into i3. @stapelberg What do you think?

**Comment by apetresc** [comment]
If the amount of configuration is the main problem here, I would actually propose making @avrelaun's 4-border style the default "only" way. It's so much more natural-looking than 3 borders, to be honest when I first started using i3 I thought the 3-border thing was a clipping bug, until I learned more about how it worked.

**Comment by avrelaun** [comment]
That would be (obviously) ok by me, I'm willing to bet most would prefer the 4-borders.

**Comment by sphaugh** [comment]
Agree that this should be a change in behavior, not new configuration. It looks/feels like a mistake that was never corrected.

**Comment by stapelberg** [comment]
With the additional proposed border, would that border consume pixels that otherwise would be available to window contents? Historically, every additional pixel we consume upsets people :).

**Comment by Airblader** [comment]
Most window managers behave like i3 here, I believe, it's just not always as obvious in the design. 

Changing this for everyone without a configuration is out of question in my eyes.

**Comment by jolange** [comment]
> Changing this for everyone without a configuration is out of question in my eyes.

I agree. In particular this would look strange with several `hide_edge_borders` options. For `hide_edge_borders both` and some layout configurations, the only visible border might be the top line.

**Comment by avrelaun** [comment]
Actually the 1 pixel border over the current titlebars would only make sense in tabbed layout but not on the stacking layout, because then the bottom titlebar would be smaller regardless of whether it is active or not. We could have both: more functionality and style without losing any pixels on the tabbed layout (where the 4 borders are the most missing), and no overlap on the stacking layout where screen real-estate is not the priority.
